src/reader/mod.rs

```diff
@@ -138,12 +161,21 @@ impl InnerIndexReader {
                 .map(SegmentReader::open)
                 .collect::<crate::Result<_>>()?
         };
+        let searcher_index_generation = Arc::new(SearcherIndexGeneration::from_segment_readers(
```

Comment:
searchable_segments above is your segment meta.

Reply:
Factored it out into new method `open()` that returns the generation & readers